redbot/cogs/mutes/mutes.py

```diff
@@ -1664,6 +1671,9 @@ async def channel_mute_user(
         new_overs: dict = {}
         move_channel = False
         new_overs.update(send_messages=False, add_reactions=False, speak=False)
+        new_overs.value &= ~(
+            self._create_public_threads ^ self._create_private_threads
+        )  # DEP-WARN Remove when updating to Discord.py 2.0
```

Comment:
This won't work with a dictionary. Looking at how `PermissionOverwrite` works, I don't think there's a way to make this work before d.py 2.0. If that's the case, this PR can be closed and we'll just wait and properly implement it after we switch.

Reply:
agreed

The only way I can see to do this completely is to patch multiple parts of discord.py, which is not the best option